### PR TITLE
Renew expired invite

### PR DESCRIPTION
**Pull request by Schparky**
In response to `/authentication` with an `invite` code that has expired, respond with a 410 and renew the invite with a new code. Also send an email to the user with the new code.

### CODE REVIEW

File: application/common/models/Authentication.php [hunk 1]
```diff
@@ -74,9 +77,15 @@ protected function authenticateByInvite($invite)
             $this->errors['invite'] = ['Invalid code.'];
             return;
         }
+
         if ( ! $invite->isValidCode()) {
```
Comment (wcjr) [inline application/common/models/Authentication.php]: it looks like an invalid code that is NOT expired will generate a new code and email, am I reading that right?

Comment (Schparky) [reply application/common/models/Authentication.php]: I was afraid of that being misleading. The previous `if` ensures that the code is a real code. `isValidCode` only checks expiration.

Comment (wcjr) [reply application/common/models/Authentication.php]: would changing the name to `isExpired` avoid the confusion?

Comment (Schparky) [reply application/common/models/Authentication.php]: done

**Pull request merged**